Src/FluentAssertions/Specialized/ExecutionTimeAssertions.cs

```diff
@@ -10,18 +12,41 @@ namespace FluentAssertions.Specialized
     /// </summary>
     public class ExecutionTimeAssertions
     {
-        private readonly TimeSpan executionTimeSpan;
-
-        private readonly string actionDescription;
+        private readonly ExecutionTime execution;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ExecutionTime"/> class.
         /// </summary>
         /// <param name="executionTime">The execution on which time must be asserted.</param>
         public ExecutionTimeAssertions(ExecutionTime executionTime)
         {
-            executionTimeSpan = executionTime.ExecutionTimeSpan;
-            actionDescription = executionTime.ActionDescription;
+            this.execution = executionTime;
+        }
+
+        /// <summary>
+        /// Checks the executing action if it satisfies a condition.
+        /// If the execution runs into an exeption, then this will rethrow it.
+        /// </summary>
+        /// <param name="condition">Condition to check un the current elapsed time.</param>
+        /// <param name="result">The expected result when polling is stopped.</param>
+        /// <param name="stopExecution">If polling is stopped then task execution is also stopped.</param>
+        /// <param name="rate">The rate at which the conition is re-checked.</param>
+        private void PollUntil(Func<TimeSpan, bool> condition, bool result, TimeSpan rate)
+        {
+            var forcedStop = false;
+            while (execution.IsRunning)
+            {
+                if (condition(execution.ElapsedTime) == result)
+                {
+                    break;
+                }
+                Task.Delay(rate).Wait();
```

Comment:
I'm not sure this is safe in all environments. I've seen lockups on AppVeyor while waiting for `Wait` call. I would just use a `Thread.Sleep` instead.

Reply:
yes, wanted to use that, originally and yes, .Result probably blocks a thread here, ill see if i can improve